src/webview/js/toggleSpoiler.js

```diff
@@ -0,0 +1,104 @@
+/* @flow strict-local */
+/* eslint-disable no-useless-return */
+
+/**
+ * Collapse an open spoiler block with animation.
+ *
+ * Does what `collapse_spoiler` in static/js/spoilers.js in the web
+ * does, only departing as appropriate to adapt to mobile.
+ */
+const collapseSpoiler = (spoiler: HTMLElement) => {
+  const spoilerHeight = spoiler.scrollHeight;
```

Comment:
Looks like this is the change we should follow here:

```diff
-   const spoiler_height = spoiler.prop('scrollHeight');
+   const spoiler_height = spoiler.height();
```

I wonder what's the best non-jQuery thing to do here. There's [`Element#clientHeight`](https://developer.mozilla.org/en-US/docs/Web/API/Element/clientHeight), or maybe [`HTMLElement#offsetHeight`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetHeight). But I'm skeptical of the first thing that occurred to me, `spoiler.style.height`; that's exactly the property we're about to write to, a few lines later.

Reply:
Ah OK, I think I've found out what "jumpy animation glitch on spoiler collapse" refers to. When you collapse a spoiler, it jumps to being slightly bigger than it was, before an animation to shrink it begins.

It appears that the amount of the jump is equal to the vertical padding in `.spoiler-content`. On mobile, at least, that's 5px on the top and 5px on the bottom. The following is sufficient to make the jump go away:

```diff
- const spoilerHeight = spoiler.scrollHeight;
+ const spoilerHeight = spoiler.scrollHeight - 10;
```

`.scrollHeight` does include the padding ([doc](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollHeight)), but my hunch is that jQuery's `.height()` does not. Sure enough; its [doc](https://api.jquery.com/height/) says:

> Note that `.height()` will always return the content height

So we need something that will just grab the content height.

Reply:
> So we need something that will just grab the content height.

According to http://youmightnotneedjquery.com/, 

```
$(el).height();
```

can be translated to

```
parseFloat(getComputedStyle(el, null).height.replace("px", ""))
```

🤷

Reply:
Great. The next thing the code does with that value is actually to turn it back to a string and add back the suffix "px" -- so that simplifies down to `getComputedStyle(el, null)`.

Reply:
And [according to MDN](https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle), that second argument is optional and `null` is equivalent to leaving it out -- so this can just be
```js
const computedHeight = getComputedStyle(spoiler);
// …
  spoiler.style.height = computedHeight;
```

Reply:
Ah, right, thanks for pointing that out!